python/test/unit/spacewalk/satellite_tools/test_yum_src.py

```diff
@@ -213,6 +214,85 @@ def test_mirror_list_arch(self):
+    @patch("spacewalk.satellite_tools.repo_plugins.yum_src.os.unlink", Mock())
+    @patch("urlgrabber.grabber.PyCurlFileObject", Mock())
+    @patch("spacewalk.common.rhnLog", Mock())
+    @patch("spacewalk.satellite_tools.repo_plugins.yum_src.fileutils.makedirs", Mock())
+    @patch("spacewalk.satellite_tools.repo_plugins.yum_src.etree.parse", MagicMock(side_effect=Exception))
+    def test_proxy_usage_with_mirrorlist(self):
+        cs = self._make_dummy_cs()
+        proxy_url = "http://proxy.example.com:8080"
+        proxy_user = "user"
+        proxy_pass = "pass"
+        cs.proxy_hostname = proxy_url
+        cs.proxy_user = proxy_user
+        cs.proxy_pass = proxy_pass
+        expected_url_list = []
+        url_list = [
+            "http://example/base/arch1/os/",
+            "http://example/",
+            "http://example.com/",
+            "https://example.org/repo/path/?token",
+            ]
+        for url in url_list:
+            if "?" in url:
+                query = "&"
+            else:
+                query = "?"
+            expected_url_list.append("{}{}proxy={}&proxyuser={}&proxypass={}".format(url,
+                                                                                     query,
+                                                                                     quote(proxy_url),
+                                                                                     proxy_user,
+                                                                                     proxy_pass
+                                                                                     ))
+        grabber_mock = Mock()
+
+        with patch("spacewalk.satellite_tools.repo_plugins.yum_src.urlgrabber.urlgrab", grabber_mock):
+            with open(os.path.join(self.repo.root, "mirrorlist.txt"), "w") as fake_list:
+                fake_list.writelines([
+                    "http://example/base/arch1/os/\n",
+                    "http://example/\n",
+                    "http://example.com/\n",
+                    "https://example.org/repo/path/?token\n",
+                ])
+            mirrors = cs._get_mirror_list(self.repo, "https://fake/repo/url")
+
+            self.assertEqual(mirrors, expected_url_list)
+
+    def test_prep_zypp_repo_url_with_proxy(self):
+        cs = self._make_dummy_cs()
+        url_list = ["http://example.com/",
+                    "https://example.com/",
+                    "https://example.org/repo/path/?token",
+                    "uln://example.com/",
+                    "uln:///channel"
+                    ]
+        uln_list = [False, False, False, True, True]
+        proxy_url = "http://proxy.example.com:8080"
+        proxy_user = "user"
+        proxy_pass = "pass"
+        cs.proxy_hostname = proxy_url
+        cs.proxy_user = proxy_user
+        cs.proxy_pass = proxy_pass
+
+        for i in range(0, len(url_list)):
+            if uln_list[i] or "?" in url_list[i]:
```

Comment:
nitpick: I would put the url and the "is uln" boolean into a tuple instead of spreading them into distinct lists, e.g.
```python
urls = [("https://example.com/", False)]
...
for url, is_uln in urls:
    if is_uln or "?" in url":
        separator = "&"
    else:
        separator = "?"
```

Reply:
Good point, I'll change that